Sources_v3/StreamChatUI/ChatChannel/ChatMessageThreadInfoView.swift

```diff
@@ -0,0 +1,132 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+open class ChatMessageThreadArrowView: View {
+    public enum Direction {
+        case toTrailing
+        case toLeading
+    }
+
+    override public class var layerClass: AnyClass {
+        CAShapeLayer.self
+    }
+
+    public var shape: CAShapeLayer {
+        layer as! CAShapeLayer
+    }
+
+    public var direction: Direction = .toTrailing {
+        didSet {
+            setNeedsDisplay()
+        }
+    }
+
+    override open func defaultAppearance() {
+        shape.strokeColor = UIColor.incomingMessageBubbleBorder.cgColor
+        shape.fillColor = nil
+        shape.lineWidth = 1.0 / UIScreen.main.scale
+    }
+
+    public var isLeftToRight: Bool {
+        let isLeftToRightWithTrailing = direction == .toTrailing && traitCollection.layoutDirection == .leftToRight
+        let isRightToLeftWithLeading = direction == .toLeading && traitCollection.layoutDirection == .rightToLeft
+        return isLeftToRightWithTrailing || isRightToLeftWithLeading
+    }
+
+    override open func draw(_ rect: CGRect) {
+        let corner: CGFloat = 16
+        let height = bounds.height
+
+        let startX = isLeftToRight ? 0 : bounds.width
+        let endX = isLeftToRight ? corner : (bounds.width - corner)
+
+        let path = CGMutablePath()
+        path.move(to: CGPoint(x: startX, y: 0))
+        path.addLine(to: CGPoint(x: startX, y: height - corner))
+        path.addQuadCurve(
+            to: CGPoint(x: endX, y: height),
+            control: CGPoint(x: startX, y: height)
+        )
+        shape.path = path
+        super.draw(rect)
+    }
+}
+
+open class ChatMessageThreadInfoView<ExtraData: UIExtraDataTypes>: View {
+    public var message: _ChatMessageGroupPart<ExtraData>? {
+        didSet { updateContentIfNeeded() }
+    }
+
+    public var onTap: (_ChatMessageGroupPart<ExtraData>?) -> Void = { _ in }
+
+    public private(set) lazy var avatarView = AvatarView().withoutAutoresizingMaskConstraints
+    public private(set) lazy var replyCountLabel: UILabel = {
+        let label = UILabel().withoutAutoresizingMaskConstraints
+        label.font = .preferredFont(forTextStyle: .footnote)
+        label.adjustsFontForContentSizeCategory = true
+        label.text = L10n.Message.Threads.reply
+        label.textColor = .messageThreadReply
```

Comment:
Isn't this the "tint" color? Everything that's blue in the designs should come from `tintColor`